#679 Set an HTML title on React pages

ahmed-arbisoft opened the pull request:
#### Pre-Flight checklist

- [x] Testing
  - [x] Changes have been manually tested

#### What are the relevant tickets?
#679 

#### What's this PR do?
Sets appropriate HTML titles on React rendered pages.

#### How should this be manually tested?
Go to any page that is rendered by React (login/enrollment/profile etc). The title of the document should change dynamically on each page.

#### Screenshots (if appropriate)
![Screenshot from 2019-07-16 16-41-04](https://user-images.githubusercontent.com/45350418/61291736-cc19b980-a7e8-11e9-9242-b12a6af68a81.png)
![Screenshot from 2019-07-16 16-40-01](https://user-images.githubusercontent.com/45350418/61291737-cc19b980-a7e8-11e9-9de1-d4e722ae8438.png)
![Screenshot from 2019-07-16 16-34-31](https://user-images.githubusercontent.com/45350418/61291739-ccb25000-a7e8-11e9-8e4b-62303a27e025.png)

noisecapella commented:
Instead of setting `document.title` in `componentDidMount`, can you try out [react-document-title](https://github.com/gaearon/react-document-title)? We used it in micromasters, for example: https://github.com/mitodl/micromasters/blob/master/static/js/components/EmploymentTab.js#L35

noisecapella commented:
Is this ready for review again?

ahmed-arbisoft commented:
Yes, sure, please go ahead. I forgot to assign the PR to you.

On Sat, 20 Jul 2019, 12:58 am George Schneeloch, <notifications@github.com>
wrote:

> Is this ready for review again?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mitodl/mitxpro/pull/851?email_source=notifications&email_token=AKZ74EUXQ7WGERQHJ6YOB7TQAIMFBA5CNFSM4IEABG5KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2MTUGA#issuecomment-513358360>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKZ74ERRE5AF73OACN7VPNTQAIMFBANCNFSM4IEABG5A>
> .
>

-- 


The content of this email and attachment(s) are confidential and intended 
solely for the recipient(s) specified in the message. If you received this 
email by error, please alert the sender by replying to this email and 
follow with its deletion, so that we can ensure such an error does not 
occur in future. If you are not the intended recipient, you are strictly 
prohibited to use, copy, disseminate or store this email or its 
attachment(s).

mbertrand commented on static/js/containers/pages/CheckoutPage.js:
```diff
@@ -1,5 +1,8 @@
+/* global SETTINGS: false */
 import React from "react"
+import DocumentTitle from "react-document-title"
+import * as Constants from "../../constants"
```
maybe just do `import CHECKOUT_PAGE_TITLE from "../../constants"` since it is the only one used on this page.  Same for the other pages.

ahmed-arbisoft replied:
Updated.

ahmed-arbisoft commented on static/js/containers/pages/CheckoutPage.js:
```diff
@@ -202,17 +209,19 @@ export class CheckoutPage extends React.Component<Props, State> {
-        updateProduct={this.updateProduct}
-        requestPending={requestPending}
-      />
+      <DocumentTitle title={`${SETTINGS.site_name} | Checkout`}>
```
Overlooked, updated.

ahmed-arbisoft commented:
@mbertrand  My bad, mistook it for an approval pending changes, I squashed the branch already. Just so you know I've incorporated all the changes you requested.
